server/start.go

```diff
@@ -184,5 +187,6 @@ func startInProcess(ctx *Context, appCreator AppCreator) (*node.Node, error) {
 	})
 
 	// run forever (the node will not be returned)
-	select {}
+	<-cancel
```

Comment:
Note, this _won't_ execute the callback/cleanup function provided to `TrapSignal`.